### ISSUE TITLE
Users with multiple associated accounts with the same email address from different providers get locked out

### ISSUE THREAD

**Issue by tgr** [paviliondev/discourse-wikimedia-auth#3]
2b6910f8 introduced a check to prevent users from logging in an `user_associated_accounts` entry with the same email address but a different UID already exists (needed since Wikimedia allows multiple user accounts with the same email address and Discourse doesn't). The check is not restricted to the mediawiki provider though, so if a user has an entry from a different provider, they get locked out.

**Comment by tgr** [comment]
Downstream task is [T244831](https://phabricator.wikimedia.org/T244831).

**Comment by angusmcleod** [comment]
@tgr Hey, just to clarify, is this plugin: 1) being used again; 2) now being used alongside other providers? If it's being used in production again, I'll need to properly review alongside the latest Discourse code.

**Comment by tgr** [comment]
Hi @angusmcleod! It's not used by the Wikimedia community (though maybe it will in the future, it's still being discussed) - the sites are in readonly mode, I ran into the issue when I wanted to log in in relation to applying some security upgrades. I figured some other group might still use it so issues should be reported here (with no expectation of support).

I think this specific issue is simple to fix so I might follow up with a pull request later (again with no expectation that anyone can offer the time to review or merge, but still seemed like the logical place to put it so any third parties can find it - of course I can find another place if you prefer).

We did not use the plugin with other providers, but having an account that was created via a different social auth provider before the plugin was turned on was enough to trigger the issue - the user had some `user_associated_accounts` records from other providers, presumably there was no way to log in with those, but the `UserAssociatedAccount.where` command still found them.
